lib/poseidon/sync_producer.rb

```diff
@@ -42,7 +42,12 @@ def send_messages(messages)
       messages_to_send = MessagesToSend.new(messages, @cluster_metadata)
 
       if refresh_interval_elapsed?
-        refresh_metadata(messages_to_send.topic_set)
+        begin
+          refresh_metadata(messages_to_send.topic_set)
+        rescue Errors::UnableToFetchMetadata => e
```

Comment:
Also, if we keep failing to get metadata and keep trying on the next attempt, do we get into some kind of cascading failure issue, where we'll just keep failing and impacting performance, or possibly the... accuracy? of the messages in each partition?